compiler/src/iree/compiler/Codegen/SPIRV/AMDConfig.cpp

```diff
@@ -38,13 +45,19 @@ namespace detail {
 LogicalResult setAMDCodeGenConfig(const spirv::TargetEnv &targetEnv,
                                   Operation *rootOp) {
   int subgroupSize = targetEnv.getResourceLimits().getSubgroupSize();
-  if (auto matmulOp = dyn_cast<linalg::MatmulOp>(rootOp)) {
-    std::array<int64_t, 2> workgroupXY = {subgroupSize / 2, 8};
-    std::array<int64_t, 3> threadMNK = {8, 4, 32};
-    return setMatmulOpConfig(matmulOp, subgroupSize, workgroupXY, threadMNK,
-                             /*useWorkgroupMemory=*/true);
+
+  if (auto linalgOp = dyn_cast<linalg::LinalgOp>(rootOp)) {
+    if (linalg::isaContractionOpInterface(linalgOp) &&
+        llvm::is_contained({2u, 3u}, linalgOp.getNumParallelLoops())) {
```

Comment:
nit: I wonder if it is worth making this check a helper? It might not be practical, so just a suggestion.